src/cluster.cpp

```diff
@@ -288,6 +287,37 @@ void cass_cluster_set_latency_aware_routing_settings(CassCluster* cluster,
   cluster->config().set_latency_aware_routing_settings(settings);
 }
 
+void cass_cluster_set_whitelist_routing(CassCluster* cluster,
+                                        cass_bool_t enabled) {
+  cluster->config().set_whitelist_routing(enabled == cass_true);
+}
+
+void cass_cluster_set_whitelist_routing_hosts(CassCluster* cluster,
+                                          const char* hosts) {
+  size_t hosts_length
+      = hosts == NULL ? 0 : strlen(hosts);
+  cass_cluster_set_whitelist_routing_hosts_n(cluster,
+                                             hosts,
+                                             hosts_length);
+}
+
+void cass_cluster_set_whitelist_routing_hosts_n(CassCluster* cluster,
+                                                const char* hosts,
+                                                size_t hosts_length) {
+  if (hosts_length == 0) {
+    cluster->config().whitelist_hosts().clear();
+  } else {
+    std::istringstream stream(std::string(hosts, hosts_length));
```

Comment:
`cass_cluster_set_contact_list()` uses this exact same functionality. It might make sense to extract this into its own function private (file static) function.